pr_agent/tools/pr_similar_issue.py

```diff
@@ -40,8 +36,7 @@ def __init__(self, issue_url: str, args: list = None):
             environment = get_settings().pinecone.environment
         except Exception:
             if not self.cli_mode:
-                repo_name, original_issue_number = self.git_provider._parse_issue_url(self.issue_url.split('=')[-1])
-                issue_main = self.git_provider.repo_obj.get_issue(original_issue_number)
+                issue_main = self.git_provider.parse_issue_url_and_get_issue(self.issue_url.split('=')[-1])
                 issue_main.create_comment("Please set pinecone api key and environment in secrets file")
             raise Exception("Please set pinecone api key and environment in secrets file")
```

Comment:
**Suggestion:** Consider using a helper function to handle the exception for the pinecone api key and environment. This will improve code readability and maintainability. [maintainability]
```suggestion
            def handle_pinecone_exception(self):
                try:
                    environment = get_settings().pinecone.environment
                except Exception:
                    if not self.cli_mode:
                        issue_main = self.git_provider.parse_issue_url_and_get_issue(self.issue_url.split('=')[-1])
                        issue_main.create_comment("Please set pinecone api key and environment in secrets file")
                    raise Exception("Please set pinecone api key and environment in secrets file")
            handle_pinecone_exception()
```